rnd/autogpt_builder/src/components/CustomNode.tsx

```diff
@@ -78,13 +80,24 @@ const CustomNode: FC<NodeProps<CustomNodeData>> = ({ data, id }) => {
     }
     current[keys[keys.length - 1]] = value;
 
+    // Remove key if value is incorrect
+    // Need to do this because otherwise schema validation 
+    // will fail for empty, constrained non-required fields
+    if (value === '' || value === null || value === undefined) {
+      delete current[keys[keys.length - 1]];
+    } else {
+      current[keys[keys.length - 1]] = value;
+    }
```

Comment:
This will cause disappearing input fields, but only depending on how `handleInputChange` is used.

If I want to replace the value in e.g. a tag field (array item), and I press `Ctrl+A` `Backspace`, the field would disappear. That's not great UX.

Instead, maybe run this when the graph is saved? A "dirty" state is allowed for WIP, but is cleaned up on save.

Reply:
Now it's cleaned on save